### ISSUE TITLE
[DEFECT] New OmniNav OffCanvas menu does not include all expected links.

### ISSUE THREAD

**Issue by kaylaziegler** [chapmanu/cascade-assets#262]
Trello story: https://trello.com/c/NWW6Ttr8

Some folders/pages are not showing up in the new omninav v2 off canvas menu, when they are expected to.

**Comment by kaylaziegler** [comment]
The cause of this was a combination of things, but it boils down to a weird Cascade/Velocity thing. These are the offending lines:
```
#set ( $metaHideSetting = $asset.metadata.getDynamicField('Hide from navigation').value )
#set ( $isVisible = $metaHideSetting != 'Yes' )
```

The problem is with the $metaHideSetting. For some reason, the value of `$asset.metadata.getDynamicField('Hide from navigation').value` was returning "" for some folders, and returning "$asset.metadata.getDynamicField('Hide from navigation').value" for others. For the latter, Velocity prints/returns the whole function call as a string literal like that when it can't find whatever you're trying to get, so in this case it couldn't find a metadata field called "Hide from navigation". So I went to one of the folders that was returning "$asset.metadata.getDynamicField('Hide from navigation').value", I checked the box "Hide from Navigation", submitted and published it, then went back and unchecked the box and submitted/published it. Then when I checked the value of $metaHideSetting again, it returned "" instead of "$asset.metadata.getDynamicField('Hide from navigation').value". Since I had first set it to "Yes", it now had a dynamic field called "Hide from navigation". Even though I unchecked it, velocity still found the value of the field, which would be empty. At least that's how I'm explaining it to myself.

Lots of ways to deal with it, I'll comment with a solution tomorrow.

**Comment by mandythomas** [comment]
that "Hide from navigation" field is a checkbox so it could be that when it's not checked the field doesn't exist, rather than existing with a value of blank. Works that way on forms on the web so maybe this is the same

**Comment by kaylaziegler** [comment]
Part of the problem was also having it in two lines, as shown in the previous comment. If "$asset.metadata.getDynamicField('Hide from navigation').value" didn't return any value, it did a weird thing where it would keep the value of $metaHideSetting from the previous iteration in the loop (the previous child asset). However, this one-liner that accomplishes the same thing does work:
`#set ( $isVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )`

**Solutions**
Adding this line inside the loop before anything else would be one way to solve the unexpected behavior we're seeing: `#set ( $metaHideSetting = '' )`

**Comment by kaylaziegler** [comment]
@mandythomas When the box is not checked, and it has never been checked, you're right that it doesn't exist for that asset. However when the box is not checked, but it was at one point, it exists with a value of ''. That's just how the system is doing it.

**Comment by tatwell** [comment]
@kaylaziegler Can you include info from last comment above as code comment (if you haven't done so already)?

**Comment by mandythomas** [comment]
I have a vague recollection about that issue from when Digital Signage was in Cascade. That was the first time I'd written for the LocatorTool instead of using the index block and structuredData. I recall that as I looped over data (any data not just hot that were check boxes) it held values from the previous loop if the value were blank or null or something. I could be wrong or it could be fixed now, so not sure if it's only because this was a checkbox, but fyi in case you need to keep that in mind as you're testing any of the fields.

**Comment by kaylaziegler** [comment]
@mandythomas Yes, I did find that that was still the case. Did a lot of testing to make sure. It's easy enough to accommodate for. Thanks!

**Comment by kaylaziegler** [comment]
Fix: PR #263
